pennylane/qaoa/mixers.py

```diff
@@ -112,3 +114,85 @@ def xy_mixer(graph):
         obs.append(qml.PauliY(node1) @ qml.PauliY(node2))
 
     return qml.Hamiltonian(coeffs, obs)
+
+
+def bit_flip_mixer(graph, n):
+    r"""Creates a bit-flip mixer Hamiltonian.
+
+    This mixer is defined as:
+
+    .. math:: H_M \ = \ \displaystyle\sum_{v \in V(G)} \frac{1}{2^{d(v)}} X_{v}
+              \displaystyle\prod_{w \in N(v)} (\mathbb{I} \ + \ (-1)^n Z_w)
+
+    where :math:`V(G)` is the set of vertices of some graph :math:`G`, :math:`d(v)` is the
+    `degree <https://en.wikipedia.org/wiki/Degree_(graph_theory)>`__ of vertex :math:`v`, and
+    :math:`N(v)` is the `neighbourhood <https://en.wikipedia.org/wiki/Neighbourhood_(graph_theory)>`__
+    of vertex :math:`v`. In addition, :math:`\mathbb{I}` is the identity operator, and :math:`X_v` and :math:`Z_v`
+    are the Pauli-Z and Pauli-X operators on vertex :math:`v`, respectively.
+
+    This mixer was introduced in *From the Quantum Approximate Optimization Algorithm
+    to a Quantum Alternating Operator Ansatz* by Stuart Hadfield, Zhihui Wang, Bryan O'Gorman,
+    Eleanor G. Rieffel, Davide Venturelli, and Rupak Biswas [`arXiv:1709.03489 <https://arxiv.org/abs/1709.03489>`__].
+
+    Args:
+         graph (nx.Graph): A graph defining the collections of wires on which the Hamiltonian acts.
+         n (int): Either :math:`0` or :math:`1`. Determines whether the mixer performs a conditional
+                  bit-flip of a vertex if all neighbouring vertices are labelled :math:`0` or :math:`1`.
```

Comment:
```suggestion
         n (int): Either :math:`0` or :math:`1`. When :math:`n=0`, a bit flip is performed on
             vertex :math:`v` only when all neighbouring nodes are in state :math:`|0\rangle`. 
             Instead, for :math:`n=1`, a bit flip is performed only when all the neighbours of 
             :math:`v` are in the state :math:`|1\rangle`.
```
Thanks for the clarification! This suggestion is longer, but I think captures what you said clearly.

Reply:
![image](https://user-images.githubusercontent.com/49409390/92942959-25e33700-f420-11ea-83cf-c1b66e2e2c3d.png)